arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -310,50 +302,50 @@ private List<Module> precompileLimitModules() {
   /** reference table modules */
   private final List<Module> refTableModules;
 
+  private final List<Module> precompileLimitModules;
+  private final List<Module> standardModules;
+  private final List<Module> standardModulesToCount;
+  private final List<Module> standardModulesToTrace;
+
+  private final Map<String, Module> modulesToCountMap;
+
   /**
    * boolean which remembers whether a {@link CreateSection} detected Failure Condition F. Gets
    * reset with every new opcode.
    */
   public boolean failureConditionForCreates = false;
 
+  public Hub(
+      final Address l2l1ContractAddress,
+      final Bytes l2l1Topic,
+      final BigInteger nonNegativeChainId,
+      final Set<String> configuredModuleKeysToCount) {
+    Preconditions.checkState(nonNegativeChainId.signum() >= 0);
+    chainId = nonNegativeChainId;
+    l2Block = new L2Block(l2l1ContractAddress, LogTopic.of(l2l1Topic));
+    l2L1Logs = new L2L1Logs(l2Block);
+    keccak = new Keccak(ecRecoverEffectiveCall, l2Block);
+    shakiraData = new ShakiraData(wcp, sha256Blocks, keccak, ripemdBlocks);
+    blockdata = new Blockdata(wcp, txnData, rlpTxn, chainId);
+    mmu = new Mmu(euc, wcp);
+    mmio = new Mmio(mmu);
+
+    refTableModules = initRefTableModules();
+    precompileLimitModules = initPrecompileLimitModules();
+    standardModules = initStandardModules();
+    standardModulesToCount = initStandardModulesToCount();
+    standardModulesToTrace = initStandardModulesToTrace();
+    modulesToCountMap = initModulesToCountMap();
+
+    modules = Stream.concat(standardModules.stream(), precompileLimitModules.stream()).toList();
```

Comment:
is modules the modules triggered during execution of the eve? If yes, we don't need the precompileLimitModule in, they are now manually triggered by other modules when needed